### PR TITLE
Drop Xorg server and libxklavier

### PR DESCRIPTION
**Pull request by jkonecny12**
This is port of the:
* https://github.com/rhinstaller/anaconda/pull/5463
* https://github.com/rhinstaller/anaconda/pull/5485
* https://github.com/rhinstaller/anaconda/pull/5470

It is first part of the Wayland migration patches to Fedora.

Resolves: https://fedoraproject.org/wiki/Changes/Anaconda_As_Native_Wayland_Application

TODO:
- [x] Add VNC migration to GRD
- [x] Add cleanup commits from (https://github.com/rhinstaller/anaconda/pull/5736)
- [x] Add overloaded print messaging fix (https://github.com/rhinstaller/anaconda/pull/5842)
- [x] PR to switch keyboard handling to systemd-localed
- [x] Patch for Lorax (https://github.com/weldr/lorax/pull/1418)
- [x] Drop temporary commit when Lorax is merged and released
- [x] Release notes
- [x] Fix existing kickstart tests for keyboard

**Ready for review**
* Tested on boot.iso - works as expected
* Tested on KDE Live - works almost as expected (there is minor bug when switching keyboard layout in system)
* Tested on Sway Live - works with additional fixes (new sway-config package needs release + https://github.com/alebastr/sway-systemd/pull/30)
* Tested on Fedora Workstation - works as expected (not impacted by this change from my testing) 

**After this merge**
* Merge https://github.com/pykickstart/pykickstart/pull/506 to remove VNC from pykickstart.
* Additional cleanups for Lorax (forwardport of https://gitlab.com/redhat/centos-stream/rpms/lorax-templates-rhel/-/merge_requests/49 and https://gitlab.com/redhat/centos-stream/rpms/lorax-templates-rhel/-/merge_requests/57)  
* Drop VNC from the DBus API backend
* Forwardport https://github.com/rhinstaller/anaconda/pull/5729

### CODE REVIEW

**Comment by jkonecny12** [comment]
Add https://github.com/rhinstaller/anaconda/pull/5842 on top of the pile.

**Comment by Conan-Kudo** [comment]
Can we please put the VNC stuff back? GRD supports VNC _and_ RDP, and there are many more high quality VNC clients than there are RDP clients for Linux. And even the first party Mac remote desktop client is VNC-based, not RDP.

**Comment by jkonecny12** [comment]
@Conan-Kudo as said a few times. I won't do that from start. This is already quite heavy patch set and getting into this VNC would make it just heavier.

Also based on the GRD maintainer reply it doesn't sound wise move in the current situation  https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/message/G4DFKOUDDDDYHNQBNYRH52XVYKWK5X2U/

**Comment by jkonecny12** [comment]
Added support for localed. Unfortunately we can't merge this until this is resolved https://pagure.io/releng/issue/12275 because it's blocking one of our new dependency.

**Comment by jkonecny12** [comment]
Also I need to do testing on Live to find how it behaves there.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix pep8 failures

**Comment by Conan-Kudo** [comment]
I don't think there is a way other than by making every compositor ship a file to turn on the feature or not. But I am not sure we should do that either, because inevitably someone will forget or not notice when it gets implemented and not ship the file and Anaconda will just not do it anyway.

**Comment by jkonecny12** [comment]
Yes, that is basically my question. What is better situation:
* Anaconda will not tell you that it can't change keyboard layouts
  * It will not break anything but user experience on unsupported systems will be bad.
* Move the responsibility on the Live spin maintainers
  * It might break some of the supported systems

OK, I guess I'll move the question to the fedora discussions to this change.

**Comment by Conan-Kudo** [comment]
I think what might make sense is a list of compositors we know support it stored as a file in Anaconda and soft-warn if the detected compositor isn't one of them.

File: pyanaconda/display.py
Comment (M4rtinK) [inline pyanaconda/display.py:90]: Yeah, this is a bit strange - we looked at this with @jstodola and it looks like copy-paste no longer works to VMs (at least in virt manager) but works fine when connecting via an RDP client. I find this strange, as I was assuming this piece of code was taking care of copy-paste working in all environments.

In any case, definitely something to look into in the future, as copy-past is often very convenient.

File: pyanaconda/display.py
Comment (M4rtinK) [inline pyanaconda/display.py:289]: "not even RDP"

File: pyanaconda/startup_utils.py
Comment (M4rtinK) [inline pyanaconda/startup_utils.py:179]: Maybe `fallback_to_tui_if_gtk_ui_is_not_available` ? Its already long, so that won't hurt & it looks better. :)

**Review by M4rtinK — APPROVED** [review]
Looks good to me & thanks a lot for making this all work on Fedora! :D Also looking very nice with localed being used. :)

File: anaconda.spec.in
Comment (jkonecny12) [reply anaconda.spec.in:259]: Fixed.

File: pyanaconda/gnome_remote_destop.py
Comment (jkonecny12) [reply pyanaconda/gnome_remote_destop.py:144]: I tested it yesterday and it happens on my VM every time it seems.

File: pyanaconda/modules/localization/localed.py
Comment (jkonecny12) [reply pyanaconda/modules/localization/localed.py:65]: Fixed.

File: pyanaconda/modules/localization/localed.py
Comment (jkonecny12) [reply pyanaconda/modules/localization/localed.py:70]: I'll change that because we are not "using" these layouts. We are just sending signals that this is what compositor is using. The point is that we should be always in control of the layouts.

File: pyanaconda/modules/localization/localed.py
Comment (jkonecny12) [reply pyanaconda/modules/localization/localed.py:74]: I'm removing the signal part. Let's just tell what happened not what we are going to do to avoid confusion.

File: pyanaconda/display.py
Comment (jkonecny12) [reply pyanaconda/display.py:289]: Fixed.

File: pyanaconda/startup_utils.py
Comment (jkonecny12) [reply pyanaconda/startup_utils.py]: Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fixed issue noted above - thanks @M4rtinK 
* Add a new comment for signal processing explaining more details issues that code is fighting with

**Comment by jkonecny12** [comment]
**UPDATED:**
* Add release notes for RDP change

**Comment by jkonecny12** [comment]
**UPDATED:**
* Resolve pep8 issues

**Comment by jkonecny12** [comment]
**UPDATE**:
* Fix typo
* Disable keyboard handling on dir and image installations

We need to disable keyboard handling on dir and image installations otherwise we would change localed configuration of the currently running system. This situation could similarly happen also on InitialSetup, however, it shouldn't be such problem there because users wants to configure keyboards there so it should be set by Anaconda in the end anyway.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix localed keyboard `options` shouldn't be removed when not specified.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fix typo in the file name as a new commit.

File: pyanaconda/gnome_remote_destop.py
Comment (M4rtinK) [reply pyanaconda/gnome_remote_destop.py]: Seems to no happen on RHEL 10 - maybe it cold be related to `systemd-reolved` not being used on RHEL ?

File: pyanaconda/modules/localization/localed.py
Comment (M4rtinK) [inline pyanaconda/modules/localization/localed.py:305]: Still something to be handled or is it for a followup PR ? :)

**Review by M4rtinK — APPROVED** [review]
Looks good to me. :)

**Comment by jkonecny12** [comment]
**UPDATE:**
* Adding one more commit to disable `fedora-cisco` repository in our containers which should resolve our blocker.

**Comment by jkonecny12** [comment]
Rebased.

**Comment by jkonecny12** [comment]
The failing KS test on all suite run is this issue https://github.com/rhinstaller/kickstart-tests/issues/1296 so it is not caused by this PR.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Add a new commit to avoid creation of the GRDServer class before we know it will be used.
** This should make the code more robust against running in unsupported environment.

File: pyanaconda/modules/localization/localed.py
Comment (jkonecny12) [reply pyanaconda/modules/localization/localed.py:305]: yes, I think it might be more appropriate but truth is not necessary.

**Review by M4rtinK — APPROVED** [review]
Check the changes for RDP startup specifically - looks good to me, thanks! :)

**Comment by jstodola** [comment]
https://github.com/rhinstaller/anaconda/pull/5729/commits/931dea595b840b1110c3b39f739ccb920412d14c should be added to this PR.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Rebased
* Add backport fix https://github.com/rhinstaller/anaconda/pull/5729

**Review by M4rtinK — APPROVED** [review]
Still looking good. :)

**Comment by KKoukiou** [comment]
We have agreed that although the Web UI tests are broken on this PR - we need to merge it - and wait for the new ISO to include the new deps for wayland for the Web UI CI to consume. 

All the efforts we did to make this work ended unsuccesfully (Manual testing of Web UI happened - it looks fine).

**Comment by rvykydal** [comment]
For the reference, these kickstar-test flakes / issues started to appear also on daily-iso after merge of the PR (used to be seen only on rhel-10 before):
https://github.com/rhinstaller/kickstart-tests/issues/1261
https://github.com/rhinstaller/kickstart-tests/issues/1296

**Pull request merged**
